src/Cupboard.Providers.Windows/WmiFacts.cs

```diff
@@ -31,18 +31,18 @@ public IEnumerable<(string Name, object Value)> GetFacts(IRemainingArguments arg
                 yield break;
             }
 
-            var cimSession = CimSession.Create(null);
-            var inst = cimSession.GetInstance("root\\cimv2", new CimInstance("Win32_OperatingSystem", "root\\cimv2"));
-
-            foreach (var prop in inst.CimInstanceProperties)
+            var keys = string.Join(',', _properties.Keys);
+            var searcher = new ManagementObjectSearcher($"SELECT {keys} FROM Win32_OperatingSystem");
+            var osDetailsCollection = searcher.Get();
+            foreach (var prop in osDetailsCollection)
             {
-                if (_properties.ContainsKey(prop.Name))
+                foreach (var property in _properties)
                 {
-                    yield return (_properties[prop.Name], prop.Value);
+                    yield return (property.Value, prop[property.Key]);
```

Comment:
FYI, this for each is necessary to get the enumeration going but there should only be returned one object.
These are not the nested for each loop your looking for.